libs/model-assessment/src/lib/ModelAssessmentDashboard/Controls/ModelOverview/RegressionDistributionChart.tsx

```diff
@@ -0,0 +1,306 @@
+}
+
+interface IRegressionDistributionChartState {
+  targetOption?: IChoiceGroupOption;
+  newlySelectedTargetOption?: IChoiceGroupOption;
+  targetFlyoutIsVisible: boolean;
+}
+
+const targetOptions: IChoiceGroupOption[] = [
+  {
+    key: JointDataset.PredictedYLabel,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .predictedY
+  },
+  {
+    key: JointDataset.TrueYLabel,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .trueY
+  },
+  {
+    key: JointDataset.RegressionError,
+    text: localization.ModelAssessment.ModelOverview.regressionTargetOptions
+      .error
+  }
+];
+
+export class RegressionDistributionChart extends React.Component<
+  IRegressionDistributionChartProps,
+  IRegressionDistributionChartState
+> {
+  public static contextType = ModelAssessmentContext;
+  public context: React.ContextType<typeof ModelAssessmentContext> =
+    defaultModelAssessmentContext;
+
+  public constructor(props: IRegressionDistributionChartProps) {
+    super(props);
+    this.state = { targetFlyoutIsVisible: false };
+  }
+
+  public componentDidMount(): void {
+    if (this.state.targetOption === undefined) {
+      const firstOption = targetOptions[0];
+      this.setState({
+        newlySelectedTargetOption: firstOption,
+        targetOption: firstOption
+      });
+    }
+  }
+
+  public async componentDidUpdate(
+    prevProps: IRegressionDistributionChartProps,
+    prevState: IRegressionDistributionChartState
+  ): Promise<void> {
+    if (
+      this.props.boxPlotState.boxPlotData.length === 0 ||
+      !_.isEqual(prevProps.cohorts, this.props.cohorts) ||
+      prevState.targetOption !== this.state.targetOption
+    ) {
+      const targetOption = this.state.targetOption
+        ? this.state.targetOption?.key.toString()
+        : targetOptions[0].key.toString();
+      const boxPlotData = await Promise.all(
+        this.props.cohorts.map((cohort: ErrorCohort, index: number) => {
+          return calculateBoxPlotDataFromErrorCohort(
+            cohort,
+            index,
+            targetOption
+          );
+        })
+      );
+      const outlierData = boxPlotData
+        .map((cohortBoxPlotData) => cohortBoxPlotData?.outliers)
+        .map((outlierValues, cohortIndex) => {
+          return outlierValues?.map((val) => [cohortIndex, val]);
+        })
+        .filter((list) => list !== undefined)
+        .reduce((list1, list2) => {
+          if (list1 === undefined) {
+            return list2 ?? [];
+          }
+          if (list2 === undefined) {
+            return list1 ?? [];
+          }
+          return list1.concat(list2);
+        }, []);
+      if (
+        !_.isEqual(boxPlotData, this.props.boxPlotState.boxPlotData) ||
+        !_.isEqual(this.props.boxPlotState.outlierData, outlierData)
+      ) {
+        this.props.onBoxPlotStateUpdate({ boxPlotData, outlierData });
+      }
+    }
+  }
+
+  public render(): React.ReactNode {
```

Comment:
Please help check reflow scenario with 200% zoom and then 360*256 screen size for newly added plot, thanks~

Reply:
It scales nicely with smaller size and zoom. 🙂